<a href="https://colab.research.google.com/github/FarhatGani/ODI_cricket_analysis/blob/main/ODI_bowlers_analysis.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# ODI Cricket Analysis 




### Importing Dataset

In [ ]:
import pandas as pd

df = pd.read_excel("ODI_cricket.xlsx", sheet_name="bowler", engine="openpyxl")
display(df.head(10))


# To ignore warning

import warnings
warnings.simplefilter(action='ignore', category=FutureWarning)
pd.options.mode.chained_assignment = None  # default='warn'

Player       Span  Mat  Inns  Balls   Runs  Wkts  \
0  M Muralitharan (Asia/ICC/SL)  1993-2011  350   341  18811  12326   534   
1             Wasim Akram (PAK)  1984-2003  356   351  18186  11812   502   
2            Waqar Younis (PAK)  1989-2003  262   258  12698   9919   416   
3          WPUJC Vaas (Asia/SL)  1994-2008  322   320  15775  11014   400   
4  Shahid Afridi (Asia/ICC/PAK)  1996-2015  398   372  17670  13632   395   
5       SM Pollock (Afr/ICC/SA)  1996-2008  303   297  15712   9631   393   
6          GD McGrath (AUS/ICC)  1993-2007  250   248  12970   8391   381   
7                   B Lee (AUS)  2000-2012  221   217  11185   8877   380   
8               SL Malinga (SL)  2004-2019  226   220  10936   9760   338   
9         A Kumble (Asia/INDIA)  1990-2007  271   265  14496  10412   337   

     Ave  Econ    SR   4   5  
0  23.08  3.93  35.2  15  10  
1  23.52  3.89  36.2  17   6  
2  23.84  4.68  30.5  14  13  
3  27.53  4.18  39.4   9   4  
4  34.51  4.62  44.7   4   9  
5  24.50  3.67  39.9  12   5  
6  22.02  3.88  34.0   9   7  
7  23.36  4.76  29.4  14   9  
8  28.87  5.35  32.3  11   8  
9  30.89  4.30  43.0   8   2

#### Meaning of each column:
1. **Player** = Name of players.<br>
2. **Span** = Career length of the player.<br>
3. **Mat** = Number of matches played by the player.<br>
4. **Inns**	 = Number of innings played by the player.<br>
5. **Balls** = Number of balls bowled by the player.<br>
6. **Runs** = Total number of runs the player gave away.<br>
7. **Wkts** = Total number of wickets taken by the player.<br>
8. **Ave** = Number of runs the player have conceded per wicket taken.<br>
9. **Econ** = Average number of runs the player have conceded per over bowled.<br>
10. **SR** = Average number of balls bowled per wicket taken.<br>
11. **4** = Number of innings in which the bowler took at least four wickets.<br>
12. **5** = Number of innings in which the bowler took at least five wickets.

### Renaming Column Names

In [ ]:
print(df.columns)

Index(['Player',   'Span',    'Mat',   'Inns',  'Balls',   'Runs',   'Wkts',
          'Ave',   'Econ',     'SR',        4,        5],
      dtype='object')


In [ ]:
df = df.rename(columns={'Mat':'Matches', 
                        'Inns':'Innings',
                        'Wkts': 'Wickets',
                        'Ave': 'Bowling_Average',
                        'Econ': 'Economy_Rate',
                        'SR': 'Strike_Rate',
                        4: "4_Wickets",
                        5: '5_Wickets'})

display(df.head())

Player       Span  Matches  Innings  Balls   Runs  \
0  M Muralitharan (Asia/ICC/SL)  1993-2011      350      341  18811  12326   
1             Wasim Akram (PAK)  1984-2003      356      351  18186  11812   
2            Waqar Younis (PAK)  1989-2003      262      258  12698   9919   
3          WPUJC Vaas (Asia/SL)  1994-2008      322      320  15775  11014   
4  Shahid Afridi (Asia/ICC/PAK)  1996-2015      398      372  17670  13632   

   Wickets  Bowling_Average  Economy_Rate  Strike_Rate  4_Wickets  5_Wickets  
0      534            23.08          3.93         35.2         15         10  
1      502            23.52          3.89         36.2         17          6  
2      416            23.84          4.68         30.5         14         13  
3      400            27.53          4.18         39.4          9          4  
4      395            34.51          4.62         44.7          4          9

### Number of Rows and Columns, Data Types and Missing Values

In [ ]:
# Number of rows
print("Number of Rows: ", df.shape[0])

# Number of columns
print("Number of Columns: ", df.shape[1])

Number of Rows:  77
Number of Columns:  12


In [ ]:
# checking for missing values and data types of each column
print(df.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 77 entries, 0 to 76
Data columns (total 12 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   Player           77 non-null     object 
 1   Span             77 non-null     object 
 2   Matches          77 non-null     int64  
 3   Innings          77 non-null     int64  
 4   Balls            77 non-null     int64  
 5   Runs             77 non-null     int64  
 6   Wickets          77 non-null     int64  
 7   Bowling_Average  77 non-null     float64
 8   Economy_Rate     77 non-null     float64
 9   Strike_Rate      77 non-null     float64
 10  4_Wickets        77 non-null     int64  
 11  5_Wickets        77 non-null     int64  
dtypes: float64(3), int64(7), object(2)
memory usage: 7.3+ KB
None


In [ ]:
## selecting columns after importing dataset

col_names = ['Player', 'Span', 'Bowling_Average', 'Economy_Rate', 'Strike_Rate']
df_new = df[col_names]
display(df_new.head())

Player       Span  Bowling_Average  Economy_Rate  \
0  M Muralitharan (Asia/ICC/SL)  1993-2011            23.08          3.93   
1             Wasim Akram (PAK)  1984-2003            23.52          3.89   
2            Waqar Younis (PAK)  1989-2003            23.84          4.68   
3          WPUJC Vaas (Asia/SL)  1994-2008            27.53          4.18   
4  Shahid Afridi (Asia/ICC/PAK)  1996-2015            34.51          4.62   

   Strike_Rate  
0         35.2  
1         36.2  
2         30.5  
3         39.4  
4         44.7

### Splitting 'Player' Column

In [ ]:
# splitting the 'Player' column to get the information about 'Country'
df_new[["Player_Name", "Country"]]= df_new['Player'].str.split("(", expand=True)
#display(df_new.head())

# removing Old 'Player' column
df_new.drop('Player', axis=1, inplace=True)
#display(df_new.head())

df_new['Country'] = df_new['Country'].str.replace(")", "")
display(df_new.head())

Span  Bowling_Average  Economy_Rate  Strike_Rate      Player_Name  \
0  1993-2011            23.08          3.93         35.2  M Muralitharan    
1  1984-2003            23.52          3.89         36.2     Wasim Akram    
2  1989-2003            23.84          4.68         30.5    Waqar Younis    
3  1994-2008            27.53          4.18         39.4      WPUJC Vaas    
4  1996-2015            34.51          4.62         44.7   Shahid Afridi    

        Country  
0   Asia/ICC/SL  
1           PAK  
2           PAK  
3       Asia/SL  
4  Asia/ICC/PAK

In [ ]:
# rearrange the columns
new_col_sequence = ['Player_Name', 'Country', 'Span', 'Bowling_Average', 'Economy_Rate', 'Strike_Rate']
df_new = df_new[new_col_sequence]

display(df_new.head())

Player_Name       Country       Span  Bowling_Average  Economy_Rate  \
0  M Muralitharan    Asia/ICC/SL  1993-2011            23.08          3.93   
1     Wasim Akram            PAK  1984-2003            23.52          3.89   
2    Waqar Younis            PAK  1989-2003            23.84          4.68   
3      WPUJC Vaas        Asia/SL  1994-2008            27.53          4.18   
4   Shahid Afridi   Asia/ICC/PAK  1996-2015            34.51          4.62   

   Strike_Rate  
0         35.2  
1         36.2  
2         30.5  
3         39.4  
4         44.7

### Player Played for ICC

In [ ]:
# ICC checking
def icc_check(x):
    if "ICC" in x:
        return "Yes"
    else:
        return "No"


In [ ]:
df_new['Played_for_ICC'] = df_new['Country'].apply(icc_check)
print(df_new['Played_for_ICC'].value_counts())

No     64
Yes    13
Name: Played_for_ICC, dtype: int64


##### Observation :
A total of **13 Player** played for **ICC**.

### Filtering Country Name

In [ ]:
# Removing Asia/ICC/Afr from the 'Country' column

df_new['Country'] = df_new['Country'].str.replace("Asia/", "")
df_new['Country'] = df_new['Country'].str.replace("/ICC", "")
df_new['Country'] = df_new['Country'].str.replace("ICC/", "")
df_new['Country'] = df_new['Country'].str.replace("Afr/", "")

display(df_new.head())

Player_Name Country       Span  Bowling_Average  Economy_Rate  \
0  M Muralitharan       SL  1993-2011            23.08          3.93   
1     Wasim Akram      PAK  1984-2003            23.52          3.89   
2    Waqar Younis      PAK  1989-2003            23.84          4.68   
3      WPUJC Vaas       SL  1994-2008            27.53          4.18   
4   Shahid Afridi      PAK  1996-2015            34.51          4.62   

   Strike_Rate Played_for_ICC  
0         35.2            Yes  
1         36.2             No  
2         30.5             No  
3         39.4             No  
4         44.7            Yes

In [ ]:
df_new['Country'].value_counts()

INDIA    13
PAK      12
AUS      10
SL        9
SA        8
NZ        8
ENG       6
WI        6
BAN       3
ZIM       1
AFG       1
Name: Country, dtype: int64

#### Observation :

• There are 11 different countries present on the dataset.<br>
• There are 10 Australian players and 3 Bangladeshi players present on the dataset.


### Number of years played by each Player

In [ ]:
# splitting the 'Span' column based on the "-"
df_new[['Debut_Year', 'Retiring_Year']] = df_new['Span'].str.split("-", expand=True)

# removing the "Span" column
df_new = df_new.drop("Span", axis=1)

display(df_new.head())

Player_Name Country  Bowling_Average  Economy_Rate  Strike_Rate  \
0  M Muralitharan       SL            23.08          3.93         35.2   
1     Wasim Akram      PAK            23.52          3.89         36.2   
2    Waqar Younis      PAK            23.84          4.68         30.5   
3      WPUJC Vaas       SL            27.53          4.18         39.4   
4   Shahid Afridi      PAK            34.51          4.62         44.7   

  Played_for_ICC Debut_Year Retiring_Year  
0            Yes       1993          2011  
1             No       1984          2003  
2             No       1989          2003  
3             No       1994          2008  
4            Yes       1996          2015

In [ ]:
#df_new.info()
df_new['Debut_Year'] = df_new['Debut_Year'].astype('int') 
df_new['Retiring_Year'] = df_new['Retiring_Year'].astype('int')

df_new['Years_Played'] = df_new['Retiring_Year'] - df_new['Debut_Year'] + 1

df_new = df_new.drop(['Debut_Year', "Retiring_Year"], axis=1)

display(df_new.head())

Player_Name Country  Bowling_Average  Economy_Rate  Strike_Rate  \
0  M Muralitharan       SL            23.08          3.93         35.2   
1     Wasim Akram      PAK            23.52          3.89         36.2   
2    Waqar Younis      PAK            23.84          4.68         30.5   
3      WPUJC Vaas       SL            27.53          4.18         39.4   
4   Shahid Afridi      PAK            34.51          4.62         44.7   

  Played_for_ICC  Years_Played  
0            Yes            19  
1             No            20  
2             No            15  
3             No            15  
4            Yes            20

In [ ]:
# Data Statistics
display(df_new.describe())

Bowling_Average  Economy_Rate  Strike_Rate  Years_Played
count        77.000000     77.000000    77.000000     77.000000
mean         28.958052      4.596753    37.909091     14.000000
std           4.826768      0.515814     6.060901      3.649081
min          18.680000      3.300000    26.100000      8.000000
25%          24.970000      4.280000    33.000000     12.000000
50%          29.290000      4.660000    37.800000     13.000000
75%          31.900000      4.920000    41.400000     16.000000
max          44.480000      5.830000    52.500000     24.000000

#### Players of longest period of time/shortest period of time/lowest economic rate/lowest strike rate/lowest bowling average

In [ ]:
df_new.sort_values("Years_Played", ascending = False).head(3)

Player_Name Country  Bowling_Average  Economy_Rate  Strike_Rate  \
72   SR Tendulkar    INDIA            44.48          5.10         52.2   
10  ST Jayasuriya       SL            36.75          4.78         46.0   
66   Shoaib Malik      PAK            39.18          4.66         50.3   

   Played_for_ICC  Years_Played  
72             No            24  
10             No            23  
66             No            21

In [ ]:
df_new.sort_values("Years_Played", ascending = True).head(5)

Player_Name Country  Bowling_Average  Economy_Rate  Strike_Rate  \
40   BKV Prasad    INDIA            32.30          4.67         41.4   
74  Rashid Khan      AFG            18.68          4.15         26.9   
73   BAW Mendis       SL            21.86          4.80         27.3   
49  Saeed Ajmal      PAK            22.72          4.18         32.6   
44   L Klusener       SA            29.95          4.70         38.2   

   Played_for_ICC  Years_Played  
40             No             8  
74             No             8  
73             No             8  
49             No             8  
44             No             9

In [ ]:
df_new.sort_values('Economy_Rate', ascending = True).head(2)

Player_Name Country  Bowling_Average  Economy_Rate  Strike_Rate  \
65    RJ Hadlee       NZ            21.56          3.30         39.1   
32  CEL Ambrose       WI            24.12          3.48         41.5   

   Played_for_ICC  Years_Played  
65             No            18  
32             No            13

In [ ]:
df_new.sort_values('Strike_Rate', ascending = True).head(2)

Player_Name Country  Bowling_Average  Economy_Rate  Strike_Rate  \
41     MA Starc      AUS            22.45          5.15         26.1   
74  Rashid Khan      AFG            18.68          4.15         26.9   

   Played_for_ICC  Years_Played  
41             No            12  
74             No             8

In [ ]:
df_new.sort_values('Bowling_Average', ascending = True).head(2)


Player_Name Country  Bowling_Average  Economy_Rate  Strike_Rate  \
74  Rashid Khan      AFG            18.68          4.15         26.9   
65    RJ Hadlee       NZ            21.56          3.30         39.1   

   Played_for_ICC  Years_Played  
74             No             8  
65             No            18

#### Observation
• On this dataset, **SR Tendulkar** from	*INDIA* played for the longest (24 years) period of time.<br>
• 4 players - BKV Prasad, Rashid Khan, BAW Mendis and Saeed Ajmal EACH played for 8 years, which is the shortest playing period presented on the dataset.<br>
• **RJ Hadlee** from *New Zealand* has the lowest Economy Rate ( 3.3 Runs per Over).<br>
• *Australian* player **MA Starc** has the lowest Strike Rate (26.1 Balls per Wicket).<br>
• And the only player from *Afganistan*, **Rashid Khan** has the lowest Bowling Average (18.68 Runs per Wicket).<br>